libsolidity/ast/ASTAnnotations.h

```diff
@@ -267,17 +269,22 @@ struct BinaryOperationAnnotation: ExpressionAnnotation
 
 enum class FunctionCallKind
 {
-	Unset,
 	FunctionCall,
 	TypeConversion,
 	StructConstructorCall
 };
 
 struct FunctionCallAnnotation: ExpressionAnnotation
 {
-	FunctionCallKind kind = FunctionCallKind::Unset;
+public:
+	[[nodiscard]] FunctionCallKind kind() const { return m_kind.value(); }
+	void setKind(FunctionCallKind _kind) { m_kind = std::move(_kind); }
```

Comment:
I like this PR :-)! But since you have constructed such nice implicit conversion and assignment semantics for ``SetOnce``, why not just keep it as a regular public struct member here?

Reply:
The original point was to encapsulate `kind`, which also allowed adding checking. Encapsulating it also allows changing/adding more conditions in the future.

Keeping it as a public member probably wouldn't solve the breakage in general though, since some usages might still need an explicit `value()` call. `anno.kind.value()` looks weird at the usage site (why isn't `kind` the kind, why do I have to call `value()` on it), whereas `anno.kind()` has a pretty clear meaning.

Reply:
Have you checked, will there be any instance where you'd need to call ``.value()``, once you allowed implicit conversions to ``T`` in ``SetOnce<T>``, and if so which and why?

Reply:
I don't think additional checks apart from "being present" and "being set once" make much sense for the annotation, since any more involved checking should happen, when the annotation is set instead - or do you have a good example in mind?

Reply:
Okay, I've checked and the implicit conversion appears to work in all cases. And no, I didn't have a good example in mind. I've pushed the change.